Task-2467 Added support for loading Acts Films content. This includes…

vichugofsl opened the pull request:
# Description
Added support for loading Acts Films content. This includes a new regex to validate the video files and logic to load files only when the folder name matches with fileset ID format

# Changelog
- Update the entity `bible_files_secondary`
- Update the entity `bible_fileset_tags` with the value name=`product_code:ACT` and description=`N2FAN/BSG_ACT`
- Create the zip using s3zipper
- Synchronize the product code data with Monday
- Created the folder: `s3://etl-development-input/FANBSGP2DV` to do testing

# Task
[User Story 2467](https://dev.azure.com/keyholesoftware/FCBH%20-%20Bible%20Brain/_workitems/edit/2467): Enable loading of Acts films

# How to test
I have used the following folder to test:
``````shell
time python3 load/DBPLoadController.py test s3://etl-development-input/ "FANBSGP2DV"

# Outcome
SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED;
START TRANSACTION;
INSERT  INTO bible_files (chapter_end, verse_end, file_name, file_size, duration, verse_sequence, hash_id, book_id, chapter_start, verse_start) VALUES ('1', NULL, 'Fang-BSG_ACT_1_stream.m3u8', '463', NULL, '1', '8b4001ee1809', 'ACT', '1', '1');
SET @Fang_BSG_ACT_1_stream_m3u8 = LAST_INSERT_ID();
INSERT  INTO bible_files (chapter_end, verse_end, file_name, file_size, duration, verse_sequence, hash_id, book_id, chapter_start, verse_start) VALUES ('1', NULL, 'Fang-BSG_ACT_1_web.mp4', '311', NULL, '1', '8b4001ee1809', 'ACT', '1', '1');
SET @Fang_BSG_ACT_1_web_mp4 = LAST_INSERT_ID();
INSERT  INTO bible_files (chapter_end, verse_end, file_name, file_size, duration, verse_sequence, hash_id, book_id, chapter_start, verse_start) VALUES ('1', NULL, 'Fang-BSG_ACT_1.mp4', '515477331', NULL, '1', '8b4001ee1809', 'ACT', '1', '1');
SET @Fang_BSG_ACT_1_mp4 = LAST_INSERT_ID();
INSERT  INTO bible_files (chapter_end, verse_end, file_name, file_size, duration, verse_sequence, hash_id, book_id, chapter_start, verse_start) VALUES ('28', '1', 'Fang-BSG_ACT_End_credits_stream.m3u8', '503', NULL, '1', '8b4001ee1809', 'ACT', '28', '1');
SET @Fang_BSG_ACT_End_credits_stream_m3u8 = LAST_INSERT_ID();
INSERT  INTO bible_files (chapter_end, verse_end, file_name, file_size, duration, verse_sequence, hash_id, book_id, chapter_start, verse_start) VALUES ('28', '1', 'Fang-BSG_ACT_End_credits_web.mp4', '341', NULL, '1', '8b4001ee1809', 'ACT', '28', '1');
SET @Fang_BSG_ACT_End_credits_web_mp4 = LAST_INSERT_ID();
INSERT  INTO bible_files (chapter_end, verse_end, file_name, file_size, duration, verse_sequence, hash_id, book_id, chapter_start, verse_start) VALUES ('28', '1', 'Fang-BSG_ACT_End_credits.mp4', '99672462', NULL, '1', '8b4001ee1809', 'ACT', '28', '1');
SET @Fang_BSG_ACT_End_credits_mp4 = LAST_INSERT_ID();
INSERT  INTO bible_files_secondary (file_type, hash_id, file_name) VALUES ('zip', '8b4001ee1809', 'FANBSGP2DV_ACT.zip');
INSERT  INTO bible_fileset_tags (description, admin_only, notes, iso, hash_id, name, language_id) VALUES ('N2FAN/BSG_ACT', '1', NULL, 'eng', '8b4001ee1809', 'product_code:ACT', '6414');
COMMIT;
EXIT

``````
- You should run the following unit test and should pass
```````shell
-> python3 load/TestFilenameParser.py test

# Outcome
Config:setConfigParametersFromProfile. profile: test, programRunning: TestFilenameParser.py
Config 'test' is loaded.

=====================================
File Type :  audio
=====================================
✅ Valid: ENGESVN2DA_B01_MAT_001.mp3 (Matched: audio99)
   - Group 1: ENGESVN2DA
   - Group 2: B01
   - Group 3: MAT
   - Group 4: 001
   - Group 5: mp3
✅ Valid: IRUNLCP1DA_B13_1TH_001_001-001_010.opus (Matched: audio100)
   - Group 1: IRUNLCP1DA
   - Group 2: B13
   - Group 3: 1TH
   - Group 4: 001
   - Group 5: 001
   - Group 6: 001
   - Group 7: 010
   - Group 8: opus
✅ Valid: SPANESV1DA_A23_GEN_005.webm (Matched: audio99)
   - Group 1: SPANESV1DA
   - Group 2: A23
   - Group 3: GEN
   - Group 4: 005
   - Group 5: webm
✅ Valid: FRENCHL1DA_A32_ZEC_010_003-011_009.mp3 (Matched: audio100)
   - Group 1: FRENCHL1DA
   - Group 2: A32
   - Group 3: ZEC
   - Group 4: 010
   - Group 5: 003
   - Group 6: 011
   - Group 7: 009
   - Group 8: mp3
✅ Valid: B01___01_Matthew_____ENGGIDN2DA.mp3 (Matched: audio101)
   - Group 1: B01
   - Group 2: 01
   - Group 3: Matthew
   - Group 4: ENGGIDN2DA
   - Group 5: mp3
✅ Valid: A02___12_Exodus_____SPANESV1DA.opus (Matched: audio101)
   - Group 1: A02
   - Group 2: 12
   - Group 3: Exodus
   - Group 4: SPANESV1DA
   - Group 5: opus
✅ Valid: B11___123_Psalms___FRENCHL1DA.webm (Matched: audio101)
   - Group 1: B11
   - Group 2: 123
   - Group 3: Psalms
   - Group 4: FRENCHL1DA
   - Group 5: webm
✅ Valid: B27___08_Revelation__ASMDPIN1DA.mp3 (Matched: audio101)
   - Group 1: B27
   - Group 2: 08
   - Group 3: Revelation
   - Group 4: ASMDPIN1DA
   - Group 5: mp3
❌ Invalid: B27____08_Revelation__ASMDPIN1DA.mp3 (No pattern matched)
❌ Invalid: ENGES_B01_MAT_001.mp3 (No pattern matched)
❌ Invalid: ENGES_B0_MAT_001.mp3 (No pattern matched)
❌ Invalid: ENGES_B01_MAT_00.mp3 (No pattern matched)
❌ Invalid: INVALID_FILENAME.mp3 (No pattern matched)

=====================================
File Type :  video
=====================================
❌ Invalid: English_KJV_JHN_1-1-18.mp4 (No pattern matched)
✅ Valid: English-KJV_JHN_1-1-18.mp4 (Matched: video1)
   - Group 1: English-KJV
   - Group 2: JHN
   - Group 3: 1
   - Group 4: 1
   - Group 5: 18
   - Group 6: mp4
❌ Invalid: English_KJV_JHN_1-1-18.mp4 (No pattern matched)
✅ Valid: English_JHN_1-1-18.mp4 (Matched: video1)
   - Group 1: English
   - Group 2: JHN
   - Group 3: 1
   - Group 4: 1
   - Group 5: 18
   - Group 6: mp4
✅ Valid: English-KJV-other-version-info_JHN_1-1-18.mp4 (Matched: video1)
   - Group 1: English-KJV-other-version-info
   - Group 2: JHN
   - Group 3: 1
   - Group 4: 1
   - Group 5: 18
   - Group 6: mp4
✅ Valid: English-other-version-info_JHN_1-1-18.mp4 (Matched: video1)
   - Group 1: English-other-version-info
   - Group 2: JHN
   - Group 3: 1
   - Group 4: 1
   - Group 5: 18
   - Group 6: mp4
✅ Valid: English other version info space_JHN_1-1-18.mp4 (Matched: video1)
   - Group 1: English other version info space
   - Group 2: JHN
   - Group 3: 1
   - Group 4: 1
   - Group 5: 18
   - Group 6: mp4
✅ Valid: English KJV other version info space_JHN_1-1-18.mp4 (Matched: video1)
   - Group 1: English KJV other version info space
   - Group 2: JHN
   - Group 3: 1
   - Group 4: 1
   - Group 5: 18
   - Group 6: mp4
❌ Invalid: Tajik_WBT_LUK_End_Credits.mp4 (No pattern matched)
❌ Invalid: English_KJV_JHN_End_credits.mp4 (No pattern matched)
✅ Valid: English-KJV-other-version-info_JHN_End_credits.mp4 (Matched: video2)
   - Group 1: English-KJV-other-version-info
   - Group 2: JHN
   - Group 3: End_credits
   - Group 4: mp4
✅ Valid: English-other-version-info_JHN_End_credits.mp4 (Matched: video2)
   - Group 1: English-other-version-info
   - Group 2: JHN
   - Group 3: End_credits
   - Group 4: mp4
❌ Invalid: English-other_version-info_JHN_End_credits.mp4 (No pattern matched)
✅ Valid: English other version space_JHN_End_credits.mp4 (Matched: video2)
   - Group 1: English other version space
   - Group 2: JHN
   - Group 3: End_credits
   - Group 4: mp4
✅ Valid: English KVJ other version space_JHN_End_credits.mp4 (Matched: video2)
   - Group 1: English KVJ other version space
   - Group 2: JHN
   - Group 3: End_credits
   - Group 4: mp4
✅ Valid: Tajik-WBT_LUK_End_Credits.mp4 (Matched: video2)
   - Group 1: Tajik-WBT
   - Group 2: LUK
   - Group 3: End_Credits
   - Group 4: mp4
✅ Valid: English-KJV_JHN_End_credits.mp4 (Matched: video2)
   - Group 1: English-KJV
   - Group 2: JHN
   - Group 3: End_credits
   - Group 4: mp4
✅ Valid: English_JHN_End_Credits.mp4 (Matched: video2)
   - Group 1: English
   - Group 2: JHN
   - Group 3: End_Credits
   - Group 4: mp4
✅ Valid: Fang-BSG_ACT_1.mp4 (Matched: video1)
   - Group 1: Fang-BSG
   - Group 2: ACT
   - Group 3: 1
   - Group 4: None
   - Group 5: None
   - Group 6: mp4
✅ Valid: Fang-BSG_ACT_End_credits.mp4 (Matched: video2)
   - Group 1: Fang-BSG
   - Group 2: ACT
   - Group 3: End_credits
   - Group 4: mp4
✅ Valid: COVENANT_SEGMENT 01 – Intro and Garden of Eden.mp4 (Matched: video5)
   - Group 1: COVENANT
   - Group 2: 01
   - Group 3:  – Intro and Garden of Eden
   - Group 4: mp4
✅ Valid: COVENANT_SEGMENT 02 – The Fall.mp4 (Matched: video5)
   - Group 1: COVENANT
   - Group 2: 02
   - Group 3:  – The Fall
   - Group 4: mp4
❌ Invalid: COVENANT - Fall.mp4 (No pattern matched)
❌ Invalid: INVALID_FILENAME.mp4 (No pattern matched)

=====================================
File Type :  text
=====================================
✅ Valid: 001GEN.usx (Matched: text3)
   - Group 1: 001
   - Group 2: GEN
   - Group 3: usx
✅ Valid: 040MAT.usx (Matched: text3)
   - Group 1: 040
   - Group 2: MAT
   - Group 3: usx
✅ Valid: 041MRK.usx (Matched: text3)
   - Group 1: 041
   - Group 2: MRK
   - Group 3: usx
❌ Invalid: 001GEN_001-010.usx (No pattern matched)
❌ Invalid: INVALID_FILENAME.usx (No pattern matched)

=====================================
Testing parse method of FilenameParser
=====================================
✅ Passed: 001GEN.usx
✅ Passed: ENGESVN2DA_B01_MAT_001.mp3
✅ Passed: IRUNLCP1DA_B13_1TH_001_001-001_010.opus
✅ Passed: English-other-version-info_JHN_End_credits.mp4
✅ Passed: English-other version info space_JHN_1-1-18.mp4
✅ Passed: English-KJV_JHN_1-1-18.mp4
✅ Passed: English-KJV_JHN_End_credits.mp4
✅ Passed: COVENANT_SEGMENT 01 – Intro and Garden of Eden.mp4
✅ Passed: Fang-BSG_ACT_1.mp4
✅ Passed: Fang-BSG_ACT_End_credits.mp4
✅ Passed: 001GEN_001-010.usx
✅ Passed: INVALID_FILENAME.mp3
✅ Passed: ENGES_B01_MAT_00.mp3

Summary:
✅ Test executed successfully.

````````

vichugofsl commented on load/UpdateDBPBibleFilesSecondary.py:
```diff
@@ -143,7 +143,7 @@ def updateBibleFilesSecondary(self, hash_id, input_fileset):
 			zip_files_map = input_fileset.zipFilesIndexedByBookId()
 			# Fetch just the Gospel book IDs
-			gospel_book_name_map = self.db.selectMap("SELECT id, notes FROM books where book_group = 'Gospels'", None)
+			gospel_book_name_map = self.db.selectMap("SELECT id, notes FROM books where book_group IN ('Gospels', 'Apostolic History')", None)
```
We need to fetch 'Gospels' and 'Apostolic History' books.

vichugofsl commented on load/UpdateDBPFilesetTables.py:
```diff
@@ -340,7 +340,7 @@ def handle_input_files_for_video(self, existing_rows, hash_id, input_fileset):
 					exists, variant_file_size = s3.get_key_info(self.config.s3_vid_bucket, f"{prefix}/{filename}")
 					if not exists:
-						print(f"WARN: missing S3 key: {key}")
+						print(f"WARN: missing S3 key: {s3_key}")
```
The s3_key is current key that we are validating

vichugofsl commented on load/FilenameParser.py:
```diff
@@ -385,6 +408,10 @@ def __init__(self, config):
 			## Example: COVENANT_SEGMENT 01 – Intro and Garden of Eden.mp4
 			FilenameRegex("video5", r'^(COVENANT)_SEGMENT\s*(\d{1,2})(.*)\.(mp4)$'),
+			## Example: Acts_Ch 28.mp4
+			FilenameRegex("video6", r'^([A-Za-z\s-]+)_((?:CH|Ch|ch)) (\d{1,3})(?:-(\d{1,2})b?-(\d{1,2}))?\.(mp4)$'),
```
This changes does not apply because it will allow the following pattern and It should not:

`Acts _ ch 28 - 1b - 18.mp4`

bradflood reviewed (changes requested):
@vichugofsl  I apologize for the confusion. The content loaded in etl-development-input was the original format and has since been changed.
The only change should be to modify the existing regex

![image](https://github.com/user-attachments/assets/9d1f1aef-5f39-4442-85c8-02464ccfcd6f)

The formats in etl-development-input should not be accepted. It should be as simple as changing the regex and adding a few test cases

bradflood commented on load/TestFilenameParser.py:
```diff
@@ -36,6 +36,11 @@
     "Tajik-WBT_LUK_End_Credits.mp4",   # Full chapter - should match video2
     "English-KJV_JHN_End_credits.mp4",   # Full chapter - should match video2
     "English_JHN_End_Credits.mp4",   # Full chapter - should match video1
+    "Acts_Ch 28.mp4",   # Full chapter - should match video6
```
all of the newly added ones should not match

vichugofsl replied:
@bradflood , I have updated the current PR to handle ACT video files in the same way as the Gospel Films. I have also added test cases for this change.

``````shell
=====================================
Testing parse method of FilenameParser
=====================================
✅ Passed: 001GEN.usx
✅ Passed: ENGESVN2DA_B01_MAT_001.mp3
✅ Passed: IRUNLCP1DA_B13_1TH_001_001-001_010.opus
✅ Passed: English-other-version-info_JHN_End_credits.mp4
✅ Passed: English-other version info space_JHN_1-1-18.mp4
✅ Passed: English-KJV_JHN_1-1-18.mp4
✅ Passed: English-KJV_JHN_End_credits.mp4
✅ Passed: COVENANT_SEGMENT 01 – Intro and Garden of Eden.mp4
✅ Passed: Fang-BSG_ACT_1.mp4
✅ Passed: Fang-BSG_ACT_End_credits.mp4
✅ Passed: 001GEN_001-010.usx
✅ Passed: INVALID_FILENAME.mp3
✅ Passed: ENGES_B01_MAT_00.mp3
``````

vichugofsl commented:
> @vichugofsl I apologize for the confusion. The content loaded in etl-development-input was the original format and has since been changed. The only change should be to modify the existing regex
> 
> ![image](https://private-user-images.githubusercontent.com/1403755/437151125-9d1f1aef-5f39-4442-85c8-02464ccfcd6f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU1MjM1MTAsIm5iZiI6MTc0NTUyMzIxMCwicGF0aCI6Ii8xNDAzNzU1LzQzNzE1MTEyNS05ZDFmMWFlZi01ZjM5LTQ0NDItODVjOC0wMjQ2NGNjZmNkNmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MjRUMTkzMzMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGRkMjFmMmY3NmQxNTU1NDkyMzUwZTE4ZDZhMGI0NDkxMTRkYTFmMmMzYzNlMzdhYmNlNmIwYzZlOWViNWI2ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.eoZ61H65netibtDtelNrRTNmFSZyEbT7oflcbG-_p2I)
> 
> The formats in etl-development-input should not be accepted. It should be as simple as changing the regex and adding a few test cases

@bradflood I have updated this PR to process ACT videos the same way as Gospel Films and added a regex that allows for missing verse values.

vichugofsl commented on load/UpdateDBPLPTSTable.py:
```diff
@@ -249,10 +249,10 @@ def updateBibleProductCode(self, inputFileset, hashId):
 			PRODUCT_CODE_VIDEO_TEMPLATE = "product_code:%s"
+			# We need to get the list of books that are allowed for the video fileset (Gospels and Apostolic History)
+			booksAllowed = self.languageReader.getGospelsAndApostolicHistoryBooks()
 
```
I think that it is not necessary because in the future we could remove one of the Gospel Film values from the database and so, we would also need to update the code base.

vichugofsl commented on load/BlimpLanguageReader.py:
```diff
@@ -78,3 +78,6 @@ def getFilesetRecords(self, filesetId):
         return record
+    # Get the list of books for the Gospels and Apostolic History group
+    def getGospelsAndApostolicHistoryBooks(self):
+       return self.service.getGospelsAndApostolicHistoryMap()
```
I think that it is not necessary because we did a similar validation in the line: https://github.com/faithcomesbyhearing/dbp-etl/pull/213/files#diff-81651d56c717d16e5219c76262404375d2cdcd555053460c68c92f1bec3a33a5R92
